include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_utils.h

```diff
@@ -669,9 +669,15 @@ struct __result_and_scratch_storage
 //A contract for future class: <sycl::event or other event, a value, sycl::buffers..., or __usm_host_or_buffer_storage>
 //Impl details: inheritance (private) instead of aggregation for enabling the empty base optimization.
 template <typename _Event, typename... _Args>
-class __future : private std::tuple<_Args...>
+class __future
 {
+  public:
+
+      using _DataTuple = std::tuple<_Args...>;
+
+  protected:
     _Event __my_event;
+    _DataTuple __my_data;
```

Comment:
I know that this is part of the experimental async API, but is this an ABI breakage?
Do we care if it is for experimental API?

What advantage do we get from moving it to a member variable from inheritance?

Reply:
In principal, the advantage is that inheritance should have place only when it's really required. I don't think that it's this place.
But I think you are correct about ABI compatibility, will fix. Thanks.

Reply:
But looks like we in any case should break ABI at this place if we want to disable `copy`-construction and `copy`-assignment for the `__future` class:
```C++
    __future(const FutureType&) = delete;
    // ...
    FutureType&
    operator=(const FutureType&) = delete;
```
Is it correct?

Reply:
@akukanov, @rarutyun what do you think about this?
Are we able to break `ABI` in the `__future` class?

Reply:
Restored original inheritance from `std::tuple`